Services/UICore/classes/Screen/PageContentProvider.php

```diff
@@ -229,8 +229,8 @@ public function getFooterModification(CalledContexts $screen_context_stack): ?Fo
 
             $footer = $f->mainControls()->footer($links, $text);
 
-            $tosWithdrawalGui = new \ilTermsOfServiceWithdrawalGUIHelper($this->dic->user());
-            $footer = $tosWithdrawalGui->modifyFooter($footer);
+            global $DIC;
+            $footer = $DIC['legalDocuments']->modifyFooter($footer);
```

Comment:
Hi @lscharmer,

I think we can get rid of this snippet if your new service implements its own `ModificationProvider`, or am I missing something? If so, I would appreciate it very much to get rid of this modification here =).

Kind regards,
@thibsy

Reply:
Thx a lot!

Reply:
@lscharmer you're right, it isn't possible to provide multiple modifications as I thought. The one with highest priority wins and replaces the lower priority modifications. But @chfsx notified me about an upcoming revision of the footer (see https://docu.ilias.de/goto_docu_wiki_wpage_7883_1357.html), which will introduce an easier mechanism to provide footer-parts. So for the time being this change LGTM, but please change `$DIC` to `$this->dic`, since the container is already available there.

Reply:
`$DIC` is replaced with `$this->dic`. Ah ok, when the footer is updated I will remove the code in another PR.